--- lib/telemetry/telemetry.c.orig      2022-11-27 10:36:36.000000000 +0000
+++ lib/telemetry/telemetry.c   2023-10-18 20:17:50.813113748 +0000
@@ -398,6 +398,12 @@
 static void *
 socket_listener(void *socket)
 {
+        struct sched_param sp;
+        /* Set the priority to the highest priority allowed. */
+        sp.sched_priority = sched_get_priority_max(SCHED_FIFO);
+        /* Set the priority of the program to the highest priority allowed. */
+       sched_setscheduler(0, SCHED_FIFO, &sp);
+
        while (1) {
                pthread_t th;
                int rc;
